fvm/transactionEnv.go

```diff
@@ -93,6 +95,7 @@ func NewTransactionEnvironment(
 		txIndex:          txIndex,
 		txID:             tx.ID(),
 		traceSpan:        traceSpan,
+		executionCtx:     context.Background(),
```

Comment:
setting executeCtx to context.Background() here enables us to reuse the same code for both ScriptEnv and TransactionEnv

Reply:
You may want to put a comment about why it is important to use non-cancellable context here.  You may also consider factoring it into a global variable to not allocate it for each transaction.